services/wallet/common/utils.go

```diff
@@ -24,3 +27,65 @@ func NetworksToChainIDs(networks []*params.Network) []uint64 {
 
 	return chainIDs
 }
+
+func ArrayContainsElement[T comparable](el T, arr []T) bool {
+	for _, e := range arr {
+		if e == el {
+			return true
+		}
+	}
+	return false
+}
+
+func ArraysWithSameElements[T comparable](ar1 []T, ar2 []T, isEqual func(T, T) bool) bool {
+	if len(ar1) != len(ar2) {
+		return false
+	}
+	for _, el := range ar1 {
+		if !ArrayContainsElement(el, ar2) {
```

Comment:
not using `isEqual`

Reply:
What do you mean `isEqual` here? I am checking if an element `el` of type `T` is in an array of elements `[]T`.

Reply:
you've got an argument `isEqual func(T, T) bool` in function `ArraysWithSameElements` which is not getting used. I'm assuming you want that for cases where you have arrays of pointers, to do a "deep" comparison instead of just comparing the pointers themselves.

Maybe the same can be achieved using https://pkg.go.dev/reflect#DeepEqual in `ArrayContainsElement` instead of `==`

Reply:
Well actually I think the whole function is unused after the rebase, you can remove `ArraysWithSameElements`

Reply:
Ahh I see what you mean, so my intention was to provide `isEqual` to `ArraysWithSameElements` and then pass it to `ArrayContainsElement` to compare any type using a custom `isEqual` function, but later when I removed it from  `ArrayContainsElement`, forgot to remove it from `ArraysWithSameElements`. 

But yes, it's not in use anymore, removed now.